src/plugins/mer/mercmakebuildconfiguration.cpp

```diff
@@ -105,10 +106,16 @@ MerCMakeBuildConfigurationFactory::MerCMakeBuildConfigurationFactory()
     addSupportedTargetDeviceType(Constants::MER_DEVICE_TYPE);
 }
 
-void MerCMakeBuildConfiguration::startBuildEngine()
+void MerCMakeBuildConfiguration::ensureBuildEngineRuns()
 {
-    MerSdkKitAspect::buildEngine(target()->kit())->virtualMachine()->connectTo(
-        Sfdk::VirtualMachine::AskStartVm|Sfdk::VirtualMachine::Block);
+    BuildEngine *const engine = MerSdkKitAspect::buildEngine(target()->kit());
+    if (engine->virtualMachine()->state() == VirtualMachine::Connected)
+        return;
+
+    bool ok;
+    execAsynchronous(std::tie(ok), std::mem_fn(&VirtualMachine::connectTo),
+            engine->virtualMachine(), VirtualMachine::AskStartVm);
+    // FIXME handle failure?
```

Comment:
An idea: override isEnabled() and disabledReason(), with something like "Could not start build engine" as the reason for being disabled?

Reply:
Noted down